Core/include/Acts/Definitions/Direction.hpp

```diff
@@ -0,0 +1,169 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+
+#include <iosfwd>
+#include <string>
+
+namespace Acts {
+
+/// The direction is always with respect to a given momentum, surface normal or
+/// other general axes
+class Direction final {
+ private:
+  enum class Value : int {
+    Negative = -1,
+    Positive = 1,
+  };
+
+ public:
+  static constexpr auto Negative = Value::Negative;
```

Comment:
Just a question: Wouldn't it make more sense to make static member functions like 

```c++
static constexpr auto Negative {
   return Direction(Value::Negative);
}
```

Because otherweise we either need to construct the Direction via `Direction(Direction::Negative)` or rely on implicit construction, right?

Reply:
yeah I also wondered if static functions would be a good alternative. to me it looked a bit odd because you need a function call to access the constants. as you said right now I rely on implicit conversions which is also not great

any opinion on that @paulgessinger ?

Reply:
Implicit conversion from a well-defined type should be ok though?

Reply:
> Implicit conversion from a well-defined type should be ok though?

In principle yes, but this can also lead to unexpected errors, e.g. for template argument deduction (no implicit conversion done, but not sure if that is a realistic use case

Another advantage of the static function (as kind of a factory function) would be that we could make the constructor private and completely keep the enum internal.

I would like the static function approach more, but I'm fine with either of both. Probably not good investement of developer time to discuss this too lengthy^^